### ISSUE TITLE
bug: error serializing or deserializing a request for response of stripe::Subscription::delete

### ISSUE THREAD

**Issue by skeptrunedev** [arlyon/async-stripe#461]
### Describe the bug

When deleting a subscription like so - `stripe::Subscription::delete(&stripe_client, &subscription_stripe_id)` - the subscription properly deletes relative to stripe externally. Meaning that after this runs for a given subscription id, the subscription is properly marked as canceled on the stripe dashboard for me and I can see the webhook fire. 

However, the error in the following scenario: 

```
.map_err(|e| {
            log::error!("Failed to cancel stripe subscription: {}", e);
})
```

regarding the Rust code is `error serializing or deserializing a request`. 

Something must be wrong regarding the way the return type is attempting to be deserialized.             

### To Reproduce

Use the `stripe::Subscription::delete` function and look at the result. 

### Expected behavior

The proper deserialized response is returned when a subscription is deleted for the the `Subscription::delete` function. 

### Code snippets

```rust
stripe::Subscription::delete(&stripe_client, &subscription_stripe_id)
        .await
        .map_err(|e| {
            log::error!("Failed to cancel stripe subscription: {}", e);
            DefaultError {
                message: "Request to stripe failed",
            }
        })?;
```
```


### OS

linux Ubuntu 22.04

### Rust version

stable-x86_64-unknown-linux-gnu

### Library version

0.26.0

### API version

2023-10-16

### Additional context

Not the biggest deal in the world because I am able to make this request manually instead, but it's not ideal. Would be great to use the crate here instead of hacking it.

**Comment by augustoccesar** [comment]
Can confirm that this is an issue when using the [`Subscription::delete`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/resources/generated/subscription.rs#L265-L267) that is on the generated folder.
The issue is that it expects a [`Deleted`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/params.rs#L67-L74) type to be returned, [which is not true for this case.
](https://stripe.com/docs/api/subscriptions/cancel)The correct way to "delete" a Stripe subscription is by canceling it with [`Subscription::cancel`. Which is correctly implemented on `subscription_ext`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/resources/subscription_ext.rs#L37-L46).

So, to simply fix your case, you can just replace
```rust
stripe::Subscription::delete(&stripe_client, &subscription_stripe_id);
```
with
```rust
stripe::Subscription::cancel(&client, &subscription.id, CancelSubscription { ..Default::default() });
```
and you should get what you are looking for.

But I guess the ideal solution would to either fix the generated code for the subscription deletion or not export it as the `Subscription::cancel` is the correct one

**Comment by augustoccesar** [comment]
> But I guess the ideal solution would to either fix the generated code for the subscription deletion or not export it as the Subscription::cancel is the correct one

I believe this will be fixed on the codegen revamp PR (https://github.com/arlyon/async-stripe/pull/452)

**Comment by skeptrunedev** [comment]
> Can confirm that this is an issue when using the [`Subscription::delete`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/resources/generated/subscription.rs#L265-L267) that is on the generated folder. The issue is that it expects a [`Deleted`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/params.rs#L67-L74) type to be returned, [which is not true for this case](https://stripe.com/docs/api/subscriptions/cancel). The correct way to "delete" a Stripe subscription is by canceling it with `Subscription::cancel`. [Which is correctly implemented on `subscription_ext`](https://github.com/arlyon/async-stripe/blob/8032120ab17a576b45faf4d9f385e8090d7d0ee3/src/resources/subscription_ext.rs#L37-L46).
> 
> So, to simply fix your case, you can just replace
> 
> ```rust
> stripe::Subscription::delete(&stripe_client, &subscription_stripe_id);
> ```
> 
> with
> 
> ```rust
> stripe::Subscription::cancel(&stripe_client, &subscription_stripe_id, CancelSubscription::new());
> ```
> 
> and you should get what you are looking for.
> 
> But I guess the ideal solution would to either fix the generated code for the subscription deletion or not export it as the `Subscription::cancel` is the correct one

Got it, that makes a lot of sense.

**Comment by mzeitlin11** [comment]
> I believe this will be fixed on the codegen revamp PR (#452)

Yep - definitely will be fixed because the `delete` method won't exist anymore! (Right now it looks like `cancel` and `delete` hit the same API endpoint, but `delete` just has the wrong return signature).

**Comment by skeptrunedev** [comment]
We did switch over to using the cancel route now, so this is no longer an active issue for us. Thanks for your help @augustoccesar
